aiohttp/connector.py

```diff
@@ -813,10 +813,13 @@ def _get_fingerprint(self, req):
         fingerprint = self._get_fingerprint(req)
 
         try:
-            hosts = await self._resolve_host(
+            # Cancelling this lookup should not cancel the underlying lookup
+            #  or else the cancel event will get broadcast to all the waiters
+            #  across all connections.
+            hosts = await asyncio.shield(self._resolve_host(
```

Comment:
So this `shield` [1] can be removed, it was put in place just for the situation that you are commenting but taking a look at your MR - and the shield implementation - I was wrong.  

[1] https://github.com/aio-libs/aiohttp/blob/master/aiohttp/connector.py#L702

Reply:
Ok was wondering about that, removed